nemo/collections/asr/data/ssl_dataset.py

```diff
@@ -0,0 +1,723 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import io
+import json
+import os
+from collections import OrderedDict
+from dataclasses import dataclass
+from math import isclose
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import torch
+from lhotse.dataset import AudioSamples
+from omegaconf import DictConfig, ListConfig, open_dict
+
+from nemo.collections.asr.data import audio_to_text, audio_to_text_dataset
+from nemo.collections.asr.parts.preprocessing.perturb import WhiteNoisePerturbation, process_augmentations
+from nemo.collections.asr.parts.preprocessing.segment import AudioSegment
+from nemo.collections.asr.parts.utils.manifest_utils import read_manifest
+from nemo.collections.common.data.dataset import ConcatDataset
+from nemo.collections.common.parts.preprocessing.manifest import get_full_path
+from nemo.core.classes import Serialization
+from nemo.utils import logging
+
+
+@dataclass
+class AudioNoiseItem:
+    sample_id: str | None = None
+    audio: torch.Tensor | None = None
+    audio_len: torch.Tensor | None = None
+    noise: torch.Tensor | None = None
+    noise_len: torch.Tensor | None = None
+    noisy_audio: torch.Tensor | None = None
+    noisy_audio_len: torch.Tensor | None = None
+
+
+@dataclass
+class AudioNoiseBatch:
+    sample_id: list | None = None
+    audio: torch.Tensor | None = None
+    audio_len: torch.Tensor | None = None
+    noise: torch.Tensor | None = None
+    noise_len: torch.Tensor | None = None
+    noisy_audio: torch.Tensor | None = None
+    noisy_audio_len: torch.Tensor | None = None
+
+
+def _parse_manifest_item(line: str, manifest_file: str) -> Dict[str, Any]:
+    """
+    Specialized function to parse the manifest file by ignoring text,
+    such that nemo dataset can save time on tokenizing text.
```

Comment:
I'm afraid it will require the same amount of modification `https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/common/parts/preprocessing/collections.py#L168`, while adding the custom parse func allows more flexibility in the future. However, I'm open to other options, please let me know~

Reply:
Its fine, we did a similar thing for manifest parsing itself for classification vs asr models.